### PR TITLE
Implement artifact inheritance support in a NuleculeComponent.

### PR DESCRIPTION
**Pull request by rtnpro**
This implements the missing feature of artifact inheritance in a ``NuleculeComponent``.

### CODE REVIEW

**Comment by rtnpro** [comment]
@dustymabe @charliedrage @vpavlin ^^

File: atomicapp/nulecule/base.py [hunk 1]
```diff
@@ -299,6 +299,11 @@ def get_artifact_paths_for_provider(self, provider_key):
                 path = Utils.sanitizePath(artifact)
                 path = os.path.join(self.basepath, path) \
                     if path[0] != '/' else path
+            elif isinstance(artifact, dict) and artifact.get('inherit') and \
+                    isinstance(artifact.get('inherit'), list):
+                inherited_provider_key = artifact.get('inherit')[0]
```
Comment (vpavlin) [inline atomicapp/nulecule/base.py]: So you check if it's a `list` and then blindly use first item? What about the rest?:-) (i.e. I am missing some `for` loop here:) )

**Comment by dustymabe** [comment]
Can you explain a little more what this "fixes". i.e. what is broken without this? I think I know but would be nice to have an explanation here for posterity.

**Comment by vpavlin** [comment]
@dustymabe You should read the spec carefully:-P https://github.com/projectatomic/nulecule/tree/master/spec#artifacts-object

**Comment by rtnpro** [comment]
@vpavlin @dustymabe,
 Fixed support for multiple items to inherit from in 94543d7

**Comment by charliedrage** [comment]
this fixes issue #323 @dustymabe

**Comment by charliedrage** [comment]
test OK now 

```
dropbox/dev/atomicapp  rtnpro-fix ✗                                                                                                                                                                                                                   2d ⚑ ◒  
▶ python -m pytest -vv -s tests/units/cli/test_cli.py::TestCli::test_run_wordpress_app
===================================================================================================================== test session starts =====================================================================================================================
platform linux2 -- Python 2.7.9 -- py-1.4.30 -- pytest-2.7.2 -- /usr/bin/python
rootdir: /home/wikus/dropbox/dev/atomicapp, inifile: 
plugins: cov
2015-10-19 09:54:17,446 - atomicapp.plugin - DEBUG - Loading providers from /home/wikus/dropbox/dev/atomicapp/atomicapp/providers
collected 10 items 

tests/units/cli/test_cli.py::TestCli::test_run_wordpress_app 2015-10-19 09:54:17,456 - atomicapp.nulecule.base - INFO - Found existing external application for aggregated-mariadb-atomicapp. Loading it.
2015-10-19 09:54:17,464 - atomicapp.plugin - DEBUG - Found provider <class 'kubernetes.KubernetesProvider'>
2015-10-19 09:54:17,464 - atomicapp.plugin - WARNING - Configuration option 'providerconfig' not found
2015-10-19 09:54:17,464 - atomicapp.plugin - DEBUG - Found provider <class 'kubernetes.KubernetesProvider'>
2015-10-19 09:54:17,464 - atomicapp.plugin - WARNING - Configuration option 'providerconfig' not found
2015-10-19 09:54:17,465 - atomicapp.plugin - DEBUG - Found provider <class 'kubernetes.KubernetesProvider'>
2015-10-19 09:54:17,465 - atomicapp.plugin - WARNING - Configuration option 'providerconfig' not found
2015-10-19 09:54:17,465 - kubernetes - DEBUG - Given config: {u'db_pass': u'MySQLPass', u'db_name': u'MySQLPass', u'db_user': u'MySQLPass', u'root_pass': u'MySQLPass', 'provider': 'kubernetes', 'namespace': 'default'}
2015-10-19 09:54:17,465 - kubernetes - INFO - Using namespace default
2015-10-19 09:54:17,465 - atomicapp.plugin - DEBUG - Found provider <class 'kubernetes.KubernetesProvider'>
2015-10-19 09:54:17,465 - atomicapp.plugin - WARNING - Configuration option 'providerconfig' not found
2015-10-19 09:54:17,465 - kubernetes - DEBUG - Given config: {'namespace': 'default', 'provider': 'kubernetes'}
2015-10-19 09:54:17,465 - kubernetes - INFO - Using namespace default
2015-10-19 09:54:17,465 - atomicapp.nulecule.main - INFO - ANSWERS: {u'aggregated-mariadb-atomicapp': {'namespace': 'default', 'provider': 'kubernetes'}, u'mariadb-atomicapp': {u'db_pass': u'MySQLPass', u'db_name': u'MySQLPass', u'db_user': u'MySQLPass', u'root_pass': u'MySQLPass'}, 'general': {'namespace': 'default', 'provider': 'kubernetes'}}
PASSED

================================================================================================================== 1 passed in 0.08 seconds ===================================================================================================================
```

@rtnpro LGTM :+1:

**Comment by dustymabe** [comment]
LGTM

**Pull request merged**